plugins/scaffolder-backend/src/scaffolder/stages/publish/github.ts

```diff
@@ -58,6 +70,19 @@ export class GithubPublisher implements PublisherBase {
     logger,
   }: PublisherOptions): Promise<PublisherResult> {
     const { owner, name } = parseGitUrl(values.storePath);
+      
+    if (this.config.credentialsProvider) {
+      this.config.token =
+        (
+          await this.config.credentialsProvider.getCredentials({
+            url: values.storePath,
+          })
+        ).token || '';
+      this.config.client = new Octokit({
```

Comment:
Probably makes sense to stop passing the client in and creating a separate one for each request here, kinda tricky to structure the code in any other way. Another option could be to try to hook the octokit auth up to the credential provider, but that seems a bit tricky.

In summary, move the pieces that change out of `this.config`, we don't want to be mutating that piece 😁

Reply:
moved the mutable variables out of `this.config` :)